modules/cudacodec/include/opencv2/cudacodec.hpp

```diff
@@ -305,16 +305,27 @@ enum DeinterlaceMode
     Adaptive = 2
 };
 
+/** @brief Output format for a decoded frame when the resolution of the source is reduced by the encoder.  In all cases the size of the output frame remains the same.
+* @param Default Use the approach adopted by cv::VideoCapture, i.e. maintain the same frame size by placing the smaller output in the top left corner.
```

Comment:
I made experiment with your test stream and VideoCapture follows frame resolution according to video:
```
Frame size: 700x400
Resolution: 700x400
Frame size: 700x400
Resolution: 700x400
Frame size: 700x400
[swscaler @ 0x55a1797812e0] Slice parameters 0, 400 are invalid
Resolution: 672x384
Frame size: 672x384
[swscaler @ 0x55a1797812e0] Slice parameters 0, 400 are invalid
Resolution: 672x384
Frame size: 672x384
Resolution: 672x384
Frame size: 672x384
Resolution: 672x384
Frame size: 672x384
Resolution: 672x384
```
Frame size: cols & rows for Mat. Resolution: CAP_PROP_FRAME_WIDTH and CAP_PROP_FRAME_HEIGHT. IMHO, it should be default behavior for consistency.

Reply:
This appears to depend on the version of FFMpeg.  If I use the prebuilt windows plugin

> --     FFMPEG:                      YES (prebuilt binaries)
> --       avcodec:                   YES (58.134.100)
> --       avformat:                  YES (58.76.100)
> --       avutil:                    YES (56.70.100)
> --       swscale:                   YES (5.9.100)
> --       avresample:                YES (4.0.0)

then I get the output you described, shown below

<details>
  <summary>avcodec: 58.134.100</summary>

```
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[swscaler @ 00000220c64f4140] Slice parameters 0, 400 are invalid
[672 x 384]
[swscaler @ 00000220c64f4140] Slice parameters 0, 400 are invalid
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
[672 x 384]
```

</details>

however when I link to a more recent version of FFmpeg, e.g.

> --     FFMPEG:                      YES (find_package)
> --       avcodec:                   YES (59.18.100)
> --       avformat:                  YES (59.16.100)
> --       avutil:                    YES (57.17.100)
> --       swscale:                   YES (6.4.100)

I get the previous output I showed in the screen shot with consistent sizes and no

> [swscaler @ 00000220c64f4140] Slice parameters 0, 400 are invalid

error message.
<details>
  <summary>avcodec: 59.18.100</summary>

```
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
[700 x 400]
```
</details>

> IMHO, it should be default behavior for consistency

Unfortunately the output from the Nvidia decoder will always be the same resolution as the original/first decoded frame (700x400) because the reconfigure functionality is designed for QOS use where you would want to maintain the same output resolution.

Reply:
I propose to rename enum items and remove reference to cv::VideoCature as soon as it behaves slightly different depending on backend.
- "default" -> "copy"
- "QoS" -> "scale" or "upscale"
It makes behaviour more obvious even without documentation.

Reply:
OK, just a thought would it be cleaner to remove `ResolutionChangeMode` completely and just default to QOS as the copy behaviour may not be a sensible default anymore (FFmpeg could change again so why copy them)?

Reply:
Ok, works for me then.